Syntax model adds

RomainGF opened the pull request.

mtp24 committed:
Syntax Model : Affectation + Variable + VariableTableau

Victor-Bonin committed:
Victor : creation des squelettes SyntaxModel

Victor-Bonin committed:
Victor : Merge avec Pourb

Victor-Bonin committed:
Victor : harmonisation du code (PAS FINI)

RomainGF committed:
modifications des includes, des opérateurs et ajout de pointeurs et du namespace std

Victor-Bonin committed:
Victor : corrections

Victor-Bonin committed:
Merge branch 'syntaxModelAdds' of https://github.com/PaulEmmanuelSotir/pld_comp into syntaxModelAdds

RomainGF committed:
Permissions sur le script de grammaire

RomainGF committed:
Untrack makefile to compile for mac

RomainGF committed:
Revert "Permissions sur le script de grammaire"

This reverts commit c2236a41202c8a981d41ae6f52b252c99b3f9976.

RomainGF committed:
Annulation du commit

RomainGF committed:
Merge branch 'master' into syntaxModelAdds

PaulEmmanuelSotir reviewed (changes requested):
- Certains attributs devraient être des pointeurs
- Il manque l'implémentation de constructeurs (ou ajouter '= default')
- Les enum pourraient être définies dans les classes qui les utilises.
- Les valeurs des enums d'opérateurs ne peuvent pas être nommées directement par leur operateur

PaulEmmanuelSotir committed:
Global SyntaxModel refactoring
Fixed some attributes types to pointers (+ added 'const' keywords when possible)
Implemented missing constructors and destructors
Moved enums to their usage class
Removed Declaration, Variable, VariableTableau, AffectationIncDecr, AtomicType and Identifier classes
Created 'Type' class
Added some missing attributes
Constant class is now a template class storing its value
Created aliases 'Identifier' and 'Include' for utils::TerminalInfo

PaulEmmanuelSotir commented:
J'ai corrigé un tas de trucs (Cf [Global SyntaxModel refactoring](https://github.com/PaulEmmanuelSotir/simplified_c_compiler/pull/1/commits/0aba4b3b6a49829176151abde039c251f13a33e7)) mais j'ai malheureusement supprimé la classe 'Variable' avant d'avoir vu la réponse sur slack et donc que l'on se mette d'accord.
Je peut la remettre mais il faut en parler avant...

PaulEmmanuelSotir committed:
Put Variable and ArrayVariable classes back

PaulEmmanuelSotir committed:
Created ArrayConstant class and fixed array_expr grammar rule

PaulEmmanuelSotir committed:
Added type attribute to ArrayConstant class
